src/repl/scala/tools/nsc/interpreter/ReplGlobal.scala

```diff
@@ -31,4 +34,14 @@ trait ReplGlobal extends Global {
       new util.AbstractFileClassLoader(virtualDirectory, loader) {}
     }
   }
+
+  override def optimizerClassPath(base: ClassPath): ClassPath = {
+    settings.outputDirs.getSingleOutput match {
+      case None => base
+      case Some(out) =>
+        // Make bytecode of previous lines available to the inliner
+        val replOutClasspath = ClassPathFactory.newClassPath(settings.outputDirs.getSingleOutput.get, settings)
+        AggregateClassPath(platform.classPath :: replOutClasspath :: Nil)
```

Comment:
Could use `AggregateClassPath.createAggregate` on this line to prevent nesting `AggregateClassPath`s